core/src/main/java/com/scalar/db/api/TwoPhaseCommitTransactionManager.java

```diff
@@ -65,9 +66,11 @@ public interface TwoPhaseCommitTransactionManager {
    * Begins a new transaction. This method is assumed to be called by a coordinator process.
    *
    * @return {@link DistributedTransaction}
+   * @throws RetriableTransactionException if something retriable situation happens. You can retry
+   *     the same transaction from the beginning in this case
    * @throws TransactionException if beginning the transaction fails
    */
-  TwoPhaseCommitTransaction begin() throws TransactionException;
+  TwoPhaseCommitTransaction begin() throws RetriableTransactionException, TransactionException;;
```

Comment:
Made this method throw `RetriableTransactionException`. Since `RetriableTransactionException` is a sub-class of `TransactionException`, it's not an incompatible changes.